src/components/Character.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { withRouter, Redirect } from 'react-router-dom';
+import './Character.css';
+
+function Character(props) {
```

Comment:
This is actually an FC (function component). One more step and you can convert others that are class based into FC + hooks.

Reply:
Character.js now uses `useHistory` hook.